integrations/shuffle.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+# Created by Shuffle, AS. <frikky@shuffler.io>.
+# Based on the Slack integration using Webhooks
+#
+# This program is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public
+# License (version 2) as published by the FSF - Free Software
+# Foundation.
+
+import contextlib
+import json
+import sys
+import time
+import os
+
+try:
+    import requests
+    from requests.auth import HTTPBasicAuth
+except Exception as e:
+    print("No module 'requests' found. Install: pip install requests")
+    sys.exit(1)
+
+# ADD THIS TO ossec.conf configuration:
+#  <integration>
+#      <name>shuffle</name>
+#      <hook_url>http://<IP>:3001/api/v1/hooks/<HOOK_ID></hook_url>
+#      <level>3</level>
+#      <alert_format>json</alert_format>
+#  </integration>
+
+# Global vars
+debug_enabled = False
+pwd = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+json_alert = {}
+now = time.strftime("%a %b %d %H:%M:%S %Z %Y")
+
+# Set paths
+log_file = f'{pwd}/logs/integrations.log'
+
+
+def main(args):
+
+    debug("# Starting")
+
+    # Read args
+    alert_file_location = args[1]
+    webhook: str = args[3]
+
+    debug("# Webhook")
+    debug(webhook)
+
+    debug("# File location")
+    debug(alert_file_location)
+
+    # Load alert. Parse JSON object.
+    try:
+        with open(alert_file_location) as alert_file:
+            json_alert = json.load(alert_file)
+    except:
+        debug("# Alert file %s doesn't exist" % alert_file_location)
+
+    debug("# Processing alert")
+    try:
+        debug(json_alert)
+    except Exception as e:
+        debug("Failed getting json_alert %s" % e)
+        sys.exit(1)
+
+    debug("# Generating message")
+    msg: str = generate_msg(json_alert)
+
+    # Check if alert is skipped
+    if isinstance(msg, str):
+        if not msg:
+            return
+
+    debug(msg)
+
+    debug("# Sending message")
+
+    send_msg(msg, webhook)
+
+
+def debug(msg):
+    if debug_enabled:
+        msg = "{0}: {1}\n".format(now, msg)
+        print(msg)
+        f = open(log_file, "a")
+        f.write(msg)
+        f.close()
+
+
+# Skips container kills to stop self-recursion
+def filter_msg(alert) -> bool:
+    # These are things that recursively happen because Shuffle starts Docker containers
+    skip_rule_ids = ["87924", "87900", "87901", "87902", "87903", "87904", "86001", "86002", "86003", "87932", "80710", "87929",
+            "87928", "5710"]
+    return not alert["rule"]["id"] in skip_rule_ids
+
+    # try:
+    #    if "docker" in alert["rule"]["description"].lower() and "
+    # msg['text'] = alert.get('full_log')
+    # except:
+    #    pass
+    # msg['title'] = alert['rule']['description'] if 'description' in alert['rule'] else "N/A"
```

Comment:
Why is this here?

Reply:
Those comments were in the original script from [the shuffle repo](https://github.com/Shuffle/Shuffle/blob/launch/functions/extensions/wazuh/custom-shuffle.py). They could be part of an alternative behavior of the script but I will remove the section because it wasn't tested and does not contribute to the clarity of the code